PurchasesCoreSwift/Identity/PurchaserInfoManager.swift

```diff
@@ -0,0 +1,198 @@
+            return
+        }
+
+        sendUpdateIfChanged(purchaserInfo: info)
+    }
+
+    @objc(purchaserInfoWithAppUserID:completionBlock:)
+    public func purchaserInfo(appUserID: String, completionBlock: ReceivePurchaserInfoBlock?) {
+        let infoFromCache = cachedPurchaserInfo(appUserID: appUserID)
+        var completionCalled = false
+
+        if let infoFromCache = infoFromCache {
+            Logger.debug(Strings.purchaserInfo.vending_cache)
+            if let completion = completionBlock {
+                completionCalled = true
+                operationDispatcher.dispatchOnMainThread {
+                    completion(infoFromCache, nil)
+                }
+            }
+        }
+
+        // Prevent calling completion twice.
+        let fetchPurchaserInfoCompletion = completionCalled ? nil : completionBlock
+
+        systemInfo.isApplicationBackgrounded { isAppBackgrounded in
+            self.fetchAndCachePurchaserInfoIfStale(appUserID: appUserID,
+                                                   isAppBackgrounded: isAppBackgrounded,
+                                                   completion: fetchPurchaserInfoCompletion)
+        }
+    }
+
+    // TODO: Ensure this is tested
+    @objc(cachedPurchaserInfoForAppUserID:)
+    public func cachedPurchaserInfo(appUserID: String) -> PurchaserInfo? {
+        guard let purchaserInfoData = deviceCache.cachedPurchaserInfoData(appUserID: appUserID) else {
+            return nil
+        }
+
+        do {
+            let maybeInfoDict = try JSONSerialization.jsonObject(with: purchaserInfoData) as? [String: Any]
+            guard let purchaserInfoDict = maybeInfoDict, let info = PurchaserInfo(data: purchaserInfoDict) else {
+                return nil
+            }
+
+            if let schema = info.schemaVersion, schema == PurchaserInfo.currentSchemaVersion {
+                return info
+            }
+        } catch {
+            Logger.warn("Unable to unmarshall PurchaserInfo from cache:\n \(error.localizedDescription)")
+        }
+
+        return nil
+    }
+
+    @objc(cachePurchaserInfo:forAppUserID:)
+    public func cache(purchaserInfo: PurchaserInfo, appUserID: String) {
+        do {
+            let jsonData = try JSONSerialization.data(withJSONObject: purchaserInfo.jsonObject())
+            deviceCache.cache(purchaserInfo: jsonData, appUserID: appUserID)
+            sendUpdateIfChanged(purchaserInfo: purchaserInfo)
+        } catch {
+            Logger.warn("Invalid JSON returned from purchaserInfo.jsonObject\n\(error.localizedDescription)")
+        }
+    }
+
+    @objc public func clearPurchaserInfoCache(forAppUserID appUserID: String) {
+        purchaserInfoCacheLock.lock()
+        deviceCache.clearPurchaserInfoCache(appUserID: appUserID)
+        lastSentPurchaserInfo = nil
+        purchaserInfoCacheLock.unlock()
+    }
+
+    // TODO: Why do we have this? It's not referenced anywhere.
```

Comment:
This is a weird one, I'm 99.9% sure I can delete, but I'll wait for somebody with more context.

Reply:
I deleted it just now, but still curious.